app/bundles/CampaignBundle/EventListener/CampaignEventSubscriber.php

```diff
@@ -68,26 +84,41 @@ public function onEventFailed(FailedEvent $event): void
         $campaign             = $failedEvent->getCampaign();
         $lead                 = $log->getLead();
         $countFailedLeadEvent = $this->eventRepository->getFailedCountLeadEvent($lead->getId(), $failedEvent->getId());
-        if ($countFailedLeadEvent < self::LOOPS_TO_FAIL) {
-            // Do not increase if under LOOPS_TO_FAIL
-            return;
-        } elseif ($countFailedLeadEvent > self::LOOPS_TO_FAIL
-            && $this->leadEventLogRepository->isLastFailed($lead->getId(), $failedEvent->getId())
-        ) {
-            // Do not increase twice
+
+        // Do not increase if under LOOPS_TO_FAIL || Do not increase twice
+        if (($countFailedLeadEvent < self::LOOPS_TO_FAIL) || ($countFailedLeadEvent > self::LOOPS_TO_FAIL
+                && $this->leadEventLogRepository->isLastFailed($lead->getId(), $failedEvent->getId())
+        )) {
             return;
         }
         // Increase if LOOPS_TO_FAIL or last success
         $failedCount   = $this->eventRepository->incrementFailedCount($failedEvent);
         $contactCount  = $campaign->getLeads()->count();
         $failedPercent = $contactCount ? ($failedCount / $contactCount) : 1;
 
-        $this->notificationHelper->notifyOfFailure($lead, $failedEvent);
+        if ($this->eventDispatcher->hasListeners(CampaignEvents::ON_CAMPAIGN_FAILURE_NOTIFY)) {
+            $this->eventDispatcher->dispatch(
+                new NotifyOfFailureEvent($lead, $failedEvent),
+                CampaignEvents::ON_CAMPAIGN_FAILURE_NOTIFY
+            );
+        }
+
+        if ($contactCount >= self::MINIMUM_CONTACTS_FOR_DISABLE
+            && $failedPercent >= self::DISABLE_CAMPAIGN_THRESHOLD
+            // Trigger only if published, if unpublished, do not trigger further notifications
+            && $campaign->isPublished()) {
+            try {
+                $this->campaignModel->transactionalCampaignUnPublish($campaign);
```

Comment:
Continuing my point from the previous comment, the campaign is unpublished here. But ```CampaignEvents::ON_CAMPAIGN_UNPUBLISH_NOTIFY``` is dispatched after this try-catch block. The campaign is always unpublished when it reaches the ```NotifyOfUnpublishSubscriber```